src/test/java/com/snowflake/kafka/connector/internal/ConnectionServiceIT.java

```diff
@@ -411,4 +424,114 @@ public void testConnectionFunction() {
+      SnowflakeSinkService service =
+          SnowflakeSinkServiceFactory.builder(
+                  conn, IngestionMethodConfig.SNOWPIPE_STREAMING, config)
+              .setRecordNumber(1)
+              .setErrorReporter(new InMemoryKafkaRecordErrorReporter())
+              .setSinkTaskContext(inMemorySinkTaskContext)
+              .addTask(tableName, topicPartition)
+              .build();
+
+      final long noOfRecords = 10;
+
+      // send regular data
+      List<SinkRecord> records =
+          TestUtils.createJsonStringSinkRecords(0, noOfRecords, tableName, 0);
+
+      service.insert(records);
+
+      TestUtils.assertWithRetry(
+          () -> service.getOffset(new TopicPartition(tableName, 0)) == noOfRecords, 5, 5);
+      channelMigrateOffsetTokenResponseDTO =
+          conn.migrateStreamingChannelOffsetToken(
+              tableName, sourceChannelName, destinationChannelName);
+      Assert.assertTrue(isChannelMigrationResponseSuccessful(channelMigrateOffsetTokenResponseDTO));
+      Assert.assertEquals(
+          OFFSET_MIGRATION_SOURCE_CHANNEL_DOES_NOT_EXIST.getStatusCode(),
+          channelMigrateOffsetTokenResponseDTO.getResponseCode());
+
+      // even after migration, it sends same offset from server side since source didnt exist
+      TestUtils.assertWithRetry(
+          () -> service.getOffset(new TopicPartition(tableName, 0)) == noOfRecords, 5, 5);
+
+      // TEST 4: Do an actual migration from new channel format to old channel Format
+      // Step 1: create a new source channel
+      // Step 2: load some data
+      // step 3: do a migration and check if destination channel has expected offset
+
+      // Ctor of TopicPartitionChannel tries to open the channel.
+      SnowflakeSinkServiceV2 snowflakeSinkServiceV2 = (SnowflakeSinkServiceV2) service;
+      TopicPartitionChannel newChannelFormatV2 =
+          new TopicPartitionChannel(
+              snowflakeSinkServiceV2.getStreamingIngestClient(),
+              topicPartition,
+              sourceChannelName,
+              tableName,
+              new StreamingBufferThreshold(10, 10_000, 1),
+              config,
+              new InMemoryKafkaRecordErrorReporter(),
+              new InMemorySinkTaskContext(Collections.singleton(topicPartition)),
+              conn,
+              conn.getTelemetryClient());
+
+      List<SinkRecord> recordsInChannelFormatV2 =
+          TestUtils.createJsonStringSinkRecords(0, noOfRecords * 2, tableName, 0);
+      recordsInChannelFormatV2.forEach(newChannelFormatV2::insertRecordToBuffer);
+
+      TestUtils.assertWithRetry(
+          () -> newChannelFormatV2.getOffsetSafeToCommitToKafka() == (noOfRecords * 2), 5, 5);
+
+      conn.migrateStreamingChannelOffsetToken(tableName, sourceChannelName, destinationChannelName);
+      TestUtils.assertWithRetry(
+          () -> service.getOffset(new TopicPartition(tableName, 0)) == (noOfRecords * 2), 5, 5);
```

Comment:
can we also send some more data after migration? to confirm that ingestion works after as well

Reply:
umm, connectionServiceIT should only test the function but let me add it in TopicPartitionChannelIT. good call!